src/preview.cpp

```diff
@@ -806,6 +806,7 @@ namespace {
                 }
                 else {
                     // FIXME: the buffer is probably copied twice, it should be optimized
+                    enforce(size_ <= static_cast<uint32_t>(io.size()), kerCorruptedMetadata);
```

Comment:
Have you considered adding this `enforce` to the place where `size_` is set?

Reply:
Yes, I tried that but the result was a bit weird. Rather than stopping with an error, `exiv2` just seemed to ignore the bad values and keep going. You can see that version of the code [here](https://github.com/kevinbackhouse/exiv2/commit/54184a3d6851f07b4d150045647ae7c6df2a7a9c). It breaks some existing tests due to the behavior that I describe.

Reply:
Well, it keeps going because that's a constructor and the only way how to abort a constructor is to throw an exception from it.

Lets keep it in the current state.